ion-schema/src/constraint.rs

```diff
@@ -682,26 +736,35 @@ impl OrderedElementsConstraint {
         type_def: &TypeDefinition,
         values_iter: &mut Peekable<Box<dyn Iterator<Item = &Element> + 'a>>,
         type_store: &TypeStore,
+        ion_path: &mut IonPath,
     ) -> ValidationResult {
         let occurs_range: Range = type_def.get_occurs_constraint("ordered_elements");
 
         // use this counter to keep track of valid values for given type_def
         let mut count: i64 = 0;
+        // use this index to keep track of Ion path for violation
+        let mut idx = 0;
```

Comment:
Subjective: If you rename this `index`, you can simplify constructing an `IndexedElement` on line 752 from
```rust
IonPathElement::IndexedElement {index: idx}
```
to
```rust
IonPathElement::IndexedElement {index}
```
It also has the nice side effect of using whole-word names, which I happen to prefer. 🙂